doc/parse/extract.go

```diff
@@ -44,10 +44,36 @@ func IsFuncInPkg(longFnName string) bool {
 	return doc != nil
 }
 
-// getShortFnName returns the name of the function, given
-// longFnName of the form:
-// github.com/adams-sarah/test2doc/example.GetWidget
+// getShortFnName returns the name of the function
+// without the package name so:
+//   github.com/user/project/package.method
+// becomes
+//   method
+// and
+//   github.com/user/project/package.(*type).method
+// becomes
+//   type.method
 func getShortFnName(longFnName string) string {
-	splitName := strings.Split(longFnName, ".")
-	return splitName[len(splitName)-1]
+
+	// drop anything before the last '/'
+	slashed := strings.Split(longFnName, "/")
+	last := slashed[len(slashed)-1]
+
+	// split the final part by period
+	dotted := strings.Split(last, ".")
+
+	// drop the first part which is the package name
+	dotted = dotted[1:]
+
+	// loop over and drop pointer references (*v) => v
```

Comment:
Alright.

So I made an example (https://gist.github.com/adams-sarah/77a24a2d56d49b0b2406448776e6c82e)

Note that in the output apib file, the function name is not even captured:
```
## /widgets/foo
### POST [POST]
```

^ POST above should be the function name, and the doc string should be below it.
Unfortunately, I don't have time to dig into why this is. Please feel free to yourself.

If you want this to get merged in, please remove this complicated pointer-removal logic. It should not affect the end result as it is (no handler name in the docs anyway).

If you figure out why the handler name is not being captured in the docs, we can revisit this.